jax/experimental/array_api/_utility_functions.py

```diff
@@ -23,3 +28,80 @@ def all(x, /, *, axis=None, keepdims=False):
 def any(x, /, *, axis=None, keepdims=False):
   """Tests whether any input array element evaluates to True along a specified axis."""
   return jax.numpy.any(x, axis=axis, keepdims=keepdims)
+
+class __array_namespace_info__:
+
+  def __init__(self):
+    self._default_dtypes = self._build_default_dtype_dict()
+    self._capabilities = {
+      "boolean indexing": True,
+      "data-dependent shapes": False,
+    }
+    self._data_types = self._build_dtype_dict()
```

Comment:
I'd lean toward computing these on-demand, rather than pre-computing them all. For example, if someone calls `device = __array_namespace_info__().default_device()`, there's no reason for us to construct a dict of all supported types in the background.